src/logging.ts

```diff
@@ -1,5 +1,158 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
 import fs = require('fs');
+import path = require('path');
+import vscode = require('vscode');
+import utils = require('./utils');
+import { ILogger } from 'vscode-jsonrpc';
+
+export enum LogLevel {
+    Verbose,
+    Normal,
+    Warning,
+    Error
+}
+
+export class Logger {
+
+    private commands: vscode.Disposable[];
+    private logChannel: vscode.OutputChannel;
+    private logFilePath: string;
+
+    public logBasePath: string;
+    public MinimumLogLevel: LogLevel = LogLevel.Normal;
+
+    constructor() {
+        this.logChannel = vscode.window.createOutputChannel("PowerShell Extension Logs");
+
+        this.commands = [
+            vscode.commands.registerCommand(
+                'PowerShell.ShowLogs',
+                () => { this.showLogPanel(); }),
+
+            vscode.commands.registerCommand(
+                'PowerShell.OpenLogFolder',
+                () => { this.openLogFolder(); })
+        ]
+    }
+
+    public getLogFilePath(baseName: string): string {
+        return path.resolve(this.logBasePath, `${baseName}.log`);
+    }
+
+    public writeAtLevel(logLevel: LogLevel, message: string, ...additionalMessages: string[]) {
+        if (logLevel >= this.MinimumLogLevel) {
+            // TODO: Add timestamp
+            this.logChannel.appendLine(message);
+            fs.appendFile(this.logFilePath, message + "\r\n");
```

Comment:
Hmm, yeah, good point.  In fact, I need to verify that this all works correctly on Linux/macOS anyway, so I'll change that before I merge.